src/x86/intel_joule_expansion.c

```diff
@@ -24,48 +25,83 @@
 
 #include <stdlib.h>
 #include <string.h>
+#include <fcntl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <fcntl.h>
+#include <sys/ioctl.h>
+#include <linux/i2c-dev.h>
+
+// Debug
+#include <syslog.h>
+#include <sys/errno.h>
 
 #include "common.h"
 #include "x86/intel_joule_expansion.h"
 
 #define PLATFORM_NAME "INTEL JOULE EXPANSION"
 
+typedef union i2c_smbus_data_union {
+    uint8_t byte;        ///< data byte
+    unsigned short word; ///< data short word
+    uint8_t block[I2C_SMBUS_BLOCK_MAX + 2];
+    ///< block[0] is used for length and one more for PEC
+} i2c_smbus_data_t;
+
+typedef struct i2c_smbus_ioctl_data_struct {
+    uint8_t read_write;     ///< operation direction
+    uint8_t command;        ///< ioctl command
+    int size;               ///< data size
+    i2c_smbus_data_t* data; ///< data
+} i2c_smbus_ioctl_data_t;
```

Comment:
We probably don't, I'll see if it can be removed with the proper includes added.